main.go

```diff
@@ -73,6 +99,19 @@ func checkFiles(w io.Writer, rootRule rules.Rule, cfg *config.Config, files []so
 		// `allowUnmatched` is explicitly set to `true` in the configuration.
 		if !result.Success || (result.Unmatched.Size() > 0 && !cfg.AllowUnmatched) {
 			problematicFiles = append(problematicFiles, file.Path)
+
+			sarifResult := rules.FormatSarif(result)
+			sarifResult.File = file.Path
+			sarifResult.Message = result.Format()
+			results = append(results, sarifResult)
+		}
+	}
+
+	if args.SarifReportPath != "" {
+		run := sarifRun(results)
+		report.AddRun(run)
+		if err := report.WriteFile(args.SarifReportPath); err != nil {
+			return fmt.Errorf("Could not write the report to the provided path:%s", args.SarifReportPath)
```

Comment:
Don't forget to wrap the initial the error here:

```suggestion
			return fmt.Errorf("Could not write the report to %s: %w", args.SarifReportPath, err)
```

Reply:
fixed